felix/bpf/bpfdefs/defs.go

```diff
@@ -14,10 +14,20 @@
 
 package bpfdefs
 
+import "os"
+
 const (
-	DefaultBPFfsPath = "/sys/fs/bpf"
-	CgroupV2Path     = "/run/calico/cgroup"
+	DefaultBPFfsPath    = "/sys/fs/bpf"
+	DefaultCgroupV2Path = "/run/calico/cgroup"
 
 	GlobalPinDir = DefaultBPFfsPath + "/tc/globals/"
 	ObjectDir    = "/usr/lib/calico/bpf"
 )
+
+func GetCgroupV2Path() string {
```

Comment:
@tomastigera I think we will want to expose this using a proper FelixConfiguration parameter rather than an environment variable, no? Is there precedence for this in the BPF data plane code?

Reply:
I think it would be tricky as this needs to get into an init container.

Reply:
Ah ok, this code isn't executed in felix proper?

It could in theory still be done via FelixConfiguration (in addition to this env var on the init container) - the operator already reads FelixConfiguration and uses it to configure various components. The flow would look like this:

- Set this field in FelixConfiguration
- tigera/operator reads FelixConfiguration, sets and env var for this container based on the value it finds

That would mean if we ever do need this value inside felix proper, it would already be available and we wouldn't need two separate configuration options. But I am not well positioned to say how likely that is.

Reply:
That sounds like a great suggestion to me :+1:

Reply:
Thanks, I will give it a shot